File: leetcode/36. selfDividingNumbers/selfDividingNumber.py
from typing import List

def selfDividingNumbers(left: int, right: int) -> List[int]:
    res = []
    for i in range(left, right+1):
        if all(i % int(x) == 0 and int(x) for x in str(i)):
            res.append(i)
    return res

File: leetcode/36. selfDividingNumbers/selfDividingNumber.go
package main

import "fmt"

func selfDividingNumbers(left int, right int) []int {
	res := make([]int, 0)
	for i := left; i < right + 1; i++ {
		j := i
		if i < 10 {
			res = append(res, i)
			continue
		}
		
		selfdivisor := true
		for i != 0 {
			cur := i % 10
			i /= 10
			if cur == 0 {
				selfdivisor = false
				break
			}
			if j % cur != 0 {
				selfdivisor = false
				break
			}
		}
		if selfdivisor {
			res = append(res, i)
		}
	}
	return res
}

func main() {
	res := selfDividingNumbers(1, 2)
	fmt.Println(res)
}